File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/InOutLineIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class InOutLineIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "inOutDocumentNumber",
            "skuIdProductId",
            "skuIdAttributeSetInstanceId",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private InOutLineId value;

    public InOutLineIdFlattenedDto()
    {
        this(new InOutLineId());
    }

    public InOutLineIdFlattenedDto(InOutLineId value)
    {
        this.value = value;
    }

    public InOutLineId toInOutLineId()
    {
        return this.value;
    }


    public String getInOutDocumentNumber()
    {
        return this.value.getInOutDocumentNumber();
    }

    public void setInOutDocumentNumber(String inOutDocumentNumber)
    {
        this.value.setInOutDocumentNumber(inOutDocumentNumber);
    }

    public String getSkuIdProductId()
    {
        return this.value.getSkuId().getProductId();
    }

    public void setSkuIdProductId(String skuIdProductId)
    {
        this.value.getSkuId().setProductId(skuIdProductId);
    }

    public String getSkuIdAttributeSetInstanceId()
    {
        return this.value.getSkuId().getAttributeSetInstanceId();
    }

    public void setSkuIdAttributeSetInstanceId(String skuIdAttributeSetInstanceId)
    {
        this.value.getSkuId().setAttributeSetInstanceId(skuIdAttributeSetInstanceId);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        InOutLineIdFlattenedDto other = (InOutLineIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/InOutLineIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\FlattenedDtoInterface;

class InOutLineIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'inOutDocumentNumber' => 'string',
            'skuIdProductId' => 'string',
            'skuIdAttributeSetInstanceId' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var InOutLineId
     */
    private $value;

    /**
     * @param InOutLineId $value
     */
    public function __construct(InOutLineId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new InOutLineId();
        }
    }

    /**
     * @return InOutLineId
     */
    public function toInOutLineId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getInOutDocumentNumber()
    {
        return $this->value->getInOutDocumentNumber();
    }

    /**
     * @param string $inOutDocumentNumber
     */
    public function setInOutDocumentNumber($inOutDocumentNumber)
    {
        $this->value->setInOutDocumentNumber($inOutDocumentNumber);
    }

    /**
     * @return string
     */
    public function getSkuIdProductId()
    {
        return $this->value->getSkuId()->getProductId();
    }

    /**
     * @param string $skuIdProductId
     */
    public function setSkuIdProductId($skuIdProductId)
    {
        $this->value->getSkuId()->setProductId($skuIdProductId);
    }

    /**
     * @return string
     */
    public function getSkuIdAttributeSetInstanceId()
    {
        return $this->value->getSkuId()->getAttributeSetInstanceId();
    }

    /**
     * @param string $skuIdAttributeSetInstanceId
     */
    public function setSkuIdAttributeSetInstanceId($skuIdAttributeSetInstanceId)
    {
        $this->value->getSkuId()->setAttributeSetInstanceId($skuIdAttributeSetInstanceId);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getInOutDocumentNumber(),
            $this->getSkuIdProductId(),
            $this->getSkuIdAttributeSetInstanceId(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return InOutLineIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setInOutDocumentNumber($pValues[0]);
        $this->setSkuIdProductId($pValues[1]);
        $this->setSkuIdAttributeSetInstanceId($pValues[2]);
        return $this;
    }

}

